manager/src/main/java/com/redhat/service/bridge/manager/connectors/ConnectorsServiceImpl.java

```diff
@@ -0,0 +1,176 @@
+package com.redhat.service.bridge.manager.connectors;
+
+import java.io.ByteArrayInputStream;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Scanner;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.openshift.cloud.api.connector.models.AddonClusterTarget;
+import com.openshift.cloud.api.connector.models.Connector;
+import com.openshift.cloud.api.connector.models.ConnectorAllOfMetadata;
+import com.openshift.cloud.api.connector.models.KafkaConnectionSettings;
+import com.redhat.service.bridge.actions.kafkatopic.KafkaTopicAction;
+import com.redhat.service.bridge.infra.models.actions.BaseAction;
+import com.redhat.service.bridge.infra.models.dto.ConnectorStatus;
+import com.redhat.service.bridge.manager.actions.connectors.ConnectorsAction;
+import com.redhat.service.bridge.manager.api.models.requests.ProcessorRequest;
+import com.redhat.service.bridge.manager.dao.ConnectorDAO;
+import com.redhat.service.bridge.manager.dao.ProcessorDAO;
+import com.redhat.service.bridge.manager.models.ConnectorEntity;
+import com.redhat.service.bridge.manager.models.Processor;
+
+@ApplicationScoped
+public class ConnectorsServiceImpl implements ConnectorsService {
+
+    @Inject
+    ObjectMapper mapper;
+
+    @Inject
+    ConnectorsApi connectorsApi;
+
+    @Inject
+    ConnectorDAO connectorDAO;
+
+    @Inject
+    ProcessorDAO processorDAO;
+
+    @ConfigProperty(name = "mc.cluster.id")
+    String mcClusterId;
+
+    @ConfigProperty(name = "kafka.bootstrap.servers")
+    String kafkaBootstrapServer;
+
+    @ConfigProperty(name = "kafka.client.id")
+    String serviceAccountId;
+
+    @ConfigProperty(name = "kafka.client.secret")
+    String serviceAccountSecret;
+
+    @Override
+    @Transactional(Transactional.TxType.MANDATORY)
```

Comment:
I think the default `REQUIRED` TxType is fine here.

Reply:
I used `MANDATORY` to avoid calling the `ConnectorService` without an active transaction from a `Processor`. In fact, we never want a connector to be created without a processor or in a different transaction. 
This is already enforced by the FK but at an API level I find it more clear. 
WDYT? I have no strong feeling about this, to me it's fine to use `REQUIRED`

Reply:
> avoid calling the ConnectorService without an active transaction from a Processor

I don't see this scenario in the code atm, unless I'm missing something?

My concern here is consistency (and explaining in comments when we deviate from the norm). For other people coming to this code from fresh without your knowledge: why is this method (and no others) annotated `MANDATORY`? Should our other `@Transactional` methods be updated to follow? When should they use `MANDATORY` or just rely on the defaults?

I think consistent and expected behaviour of our code should be one of our goals so that's why I'm +1 to rely on the defaults unless it really doesn't make sense, and when it doesn't make sense, leave a comment in the code as to why it's been done that way.

Reply:
That's exactly the point, you won't find a case when `ConnectorService` is called without an active transaction because it's forbidden by the API - that's the meaning of `MANDATORY`. It's a way to explicitly said it's forbidden to call this method without an active transaction. 

For consistency, I'd avoid using `REQUIRED` for services that needs to be called from other services (and never alone). This prevents users from using it without an active transaction.